src/pages/runs/[date].tsx

```diff
@@ -0,0 +1,81 @@
+import Layout from "@/components/Layout";
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { isMatch } from "date-fns";
+import { graphql, Link, PageProps } from "gatsby";
+import * as React from "react";
+
+const RunsPage: React.FC<PageProps<Queries.RunsPageQuery>> = ({
+  data,
+  params,
+}) => {
+  const drpdPages = data.allDrpdJson.edges;
+  const date = params.date;
+
+  return (
+    <Layout date={date}>
+      {isMatch(date, "yyyy-MM-dd") ? (
+        <div className="container mx-auto py-8">
+          <h1 className="text-3xl font-bold mb-6">
+            Available DRPD Pages for {date}
+          </h1>
+          <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-3">
+            {drpdPages
+              .filter(({ node }) => node.date === date)
+              .map(({ node }) => (
+                <Card key={node.uuid}>
+                  <CardHeader>
+                    <CardTitle>{node.title}</CardTitle>
+                  </CardHeader>
+                  <CardContent>
+                    <p className="text-sm text-muted-foreground">
+                      Date: {date}
+                    </p>
+                    <p className="text-sm text-muted-foreground">
+                      Authors: {node.authors.join(", ")}
+                    </p>
+                  </CardContent>
+                  <CardFooter className="flex justify-between">
+                    <Button asChild variant="outline">
+                      <Link to={`./${node.label}/detailed`}>Detailed</Link>
+                    </Button>
+                    <Button asChild variant="outline">
+                      <Link to={`./${node.label}/follow-along`}>
+                        Follow Along
+                      </Link>
+                    </Button>
+                  </CardFooter>
+                </Card>
+              ))}
+          </div>
+        </div>
+      ) : (
+        <div></div>
+      )}
+    </Layout>
+  );
+};
+
+export const query = graphql`
```

Comment:
Is this actually being used?

Reply:
do you mean the query? yes

the component uses the query result (which is every run data) and filter only the runs of that date to display

previously I used the createPages api to add variable to the query and only query for runs of that date but when I did that pages for the dates without a run data were not generated, so I had to use dynamic routes
I was unable to find a way to use variable in a query for dynamic routes so it just queries for every run

Reply:
Ohhh so it's a gatsby thing, I was wondering where the query was actually being called but instead it's hidden and the result data is passed to the component, very cool.